File: test/before_train/trainset_and_testset/java-py/HumanEval_81_numerical_letter_grade.py
from typing import List

def numerical_letter_grade(grades: List[float]) -> List[str]:
    letterGrade = []
    for grade in grades:
        if grade == 4.0f:
            letterGrade.append("A+")
        elif grade == 3.7f:
            letterGrade.append("A")
        elif grade == 3.3f:
            letterGrade.append("A-")
        elif grade == 3.0f:
            letterGrade.append("B+")
        elif grade == 2.7f:
            letterGrade.append("B")
        elif grade == 2.3f:
            letterGrade.append("B-")
        elif grade == 2.0f:
            letterGrade.append("C+")
        elif grade == 1.7f:
            letterGrade.append("C")
        elif grade == 1.3f:
            letterGrade.append("C-")
        elif grade == 1.0f:
            letterGrade.append("D+")
        elif grade == 0.7f:
            letterGrade.append("D")
        elif grade == 0.0f:
            letterGrade.append("D-")
        else:
            letterGrade.append("E")
    return letterGrade

def check(candidate):
    assert candidate([4.0, 3, 1.7, 2, 3.5]) == ['A+', 'B', 'C-', 'C', 'A-']
    assert candidate([1.2]) == ['D+']
    assert candidate([0.5]) == ['D-']
    assert candidate([0.0]) == ['E']
    assert candidate([1.0, 0.3, 1.5, 2.8, 3.3]) == ['D', 'D-', 'C-', 'B', 'B+']
    assert candidate([0.0, 0.7]) == ['E', 'D-']

def test_check():
    check(numerical_letter_grade)

test_check()

File: dpo_train/generation_output/py-cpp/HumanEval_81_numerical_letter_grade.cpp
### Python version

from typing import List

def numerical_letter_grade(grades: List[float]) -> List[str]:

   
    letter_grade = []
    for gpa in grades:
        if gpa == 4.0:
            letter_grade.append("A+")
        elif gpa > 3.7:
            letter_grade.append("A")
        elif gpa > 3.3:
            letter_grade.append("A-")
        elif gpa > 3.0:
            letter_grade.append("B+")
        elif gpa > 2.7:
            letter_grade.append("B")
        elif gpa > 2.3:
            letter_grade.append("B-")
        elif gpa > 2.0:
            letter_grade.append("C+")
        elif gpa > 1.7:
            letter_grade.append("C")
        elif gpa > 1.3:
            letter_grade.append("C-")
        elif gpa > 1.0:
            letter_grade.append("D+")
        elif gpa > 0.7:
            letter_grade.append("D")
        elif gpa > 0.0:
            letter_grade.append("D-")
        else:
            letter_grade.append("E")
    return letter_grade

### C++ version

#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
std::vector<std::string> numerical_letter_grade(std::vector<float> grades) {
    std::vector<std::string> letter_grade;
    for (auto g : grades) {
        letter_grade.push_back(numerical_letter_grade(g));
    }
    return letter_grade;
}

std::string numerical_letter_grade(float grade) {
    std::stringstream ss;
    ss << grade;
    return ss.str();
}

std::vector<std::string> numerical_letter_grade(std::vector<float> grades) {
    std::vector<std::string> letter_grade;
    for (auto g : grades) {
        letter_grade.push_back(numerical_letter_grade(g));
    }
    return letter_grade;
}

std::string numerical_letter_grade(std::string grade) {
    std::stringstream ss;
    ss << grade;
    return ss.str();
}

std::vector<std::string> numerical_letter_grade(std::vector<std::string> grades) {
    std::vector<std::string> letter_grade;
    for (auto g : grades) {
        letter_grade.push_back(numerical_letter_grade(g));
    }
    return letter_grade;
}

std::string numerical_letter_grade(std::string grade) {
    std::stringstream ss;
    ss << grade;
    return ss.str();
}

std::vector<std::string> numerical_letter_grade(std::vector<std::string> grades) {
    std::vector<std::string> letter_grade;
    for (auto g : grades) {
        letter_grade.push_back(numerical_letter_grade(g));
    }
    return letter_grade;
}

std::string numerical_letter_grade(std::string grade) {
    std::stringstream ss;
    ss << grade;
    return ss.str();
}

std::vector<std::string> numerical_letter_grade(std::vector<std::string> grades) {
    std::vector<std::string> letter_grade;
    for (auto g : grades) {
        letter_grade.push_back(numerical_letter_grade(g));
    }
    return letter_grade;
}

std::string numerical_letter_grade(std::string grade) {
    std::stringstream ss;
    ss << grade;
    return ss.str();
}

std::vector<std::string> numerical_letter_